### PR TITLE
RIB archiving First pass.

### PR DESCRIPTION
**Pull request by MWY3510**
Allows the use of RIB archives for single frames. Also allows the use of correction matrix for particles in the future (currently works for archives.) In addition to all that empties now are collected by their own functions to determine if they are carrying RIB archive references. So in short you can export and read any renderable object using RIB archives attached again to any type of renderable object (and empties.)

### CODE REVIEW

**Comment by MWY3510** [comment]
Let me know if there are any questions.

File: operators.py [hunk 1]
```diff
@@ -207,24 +207,75 @@ def invoke(self, context, event=None):
         return {'FINISHED'}
 
 class ExportRIBObject(bpy.types.Operator):
-    bl_idname = "object.export_rib_archive"
+    bl_idname = "export.export_rib_archive"
     bl_label = "Export Object as RIB Archive."
     bl_description = "Export single object as a RIB archive for use in other blend files or for other uses."
-    def invoke(self, context, event=None):
-        print("Exporting all the RIB!!" + str(context.active_object))
```
Comment (MWY3510) [inline operators.py]: print statement removed.

**Comment by bsavery** [comment]
So I have a thought... there's a few places you're passing in this "correction_matrix"  Isn't this always going to be ob.matrix_world?  We already have ob there....

So maybe if so, could we do something like pass in do_local_matrix=False by default, and if true then do what you do just with ob.matrix_world

**Comment by MWY3510** [comment]
Sure that would work. Also I do use an inverted version of the object's matrix for resetting the particles to local space. I can do something for that also. Thanks.

**Comment by bsavery** [comment]
Yeah just might make it a bit cleaner.

On Wednesday, March 2, 2016, MWY3510 <notifications@github.com> wrote:

> Sure that would work. Also I do use an inverted version of the object's
> matrix for resetting the particles to local space. I can do something for
> that also. Thanks.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bsavery/PRMan-for-Blender/pull/207#issuecomment-191525345>
> .
>


-- 
brian.savery@gmail.com
508-274-8700

**Comment by MWY3510** [comment]
No soon said then done (or something like that.) Keep the fantastic feedback coming.

**Pull request merged**